### PR TITLE
[Core] update deprecated interfaces

### PR DESCRIPTION
**Pull request by philbucher**
This is the first PR for updating the deprecated interfaces, first in the core

### CODE REVIEW

File: kratos/includes/condition.h [hunk 1]
```diff
@@ -331,53 +331,47 @@ class Condition : public GeometricalObject
...
     }
     // KRATOS_DEPRECATED_MESSAGE("This is legacy version, please add the missing \"const\"")
     virtual void GetValuesVector(Vector& values, int Step = 0)
     {
-        if (values.size() != 0) {
-            values.resize(0, false);
-        }
+        const auto& r_const_this = *this;
+        r_const_this.GetValuesVector(values, Step);
     }
```
Comment (rubenzorrilla) [inline kratos/includes/condition.h:343]: As far as I understand, this means that the implementation must be always in the `const` version of the `GetValuesVector`.

Taking this into account, I'm wondering if we should make the method non-virtual.

File: kratos/includes/condition.h [hunk 2]
```diff
@@ -331,53 +331,47 @@ class Condition : public GeometricalObject
...
     // KRATOS_DEPRECATED_MESSAGE("This is legacy version, please add the missing \"const\"")
     virtual void GetFirstDerivativesVector(Vector& values, int Step = 0)
     {
-        if (values.size() != 0) {
-            values.resize(0, false);
-        }
+        const auto& r_const_this = *this;
+        r_const_this.GetFirstDerivativesVector(values, Step);
     }
```
Comment (rubenzorrilla) [inline kratos/includes/condition.h:359]: Same comment.

File: kratos/includes/condition.h [hunk 3]
```diff
@@ -331,53 +331,47 @@ class Condition : public GeometricalObject
...
     // KRATOS_DEPRECATED_MESSAGE("This is legacy version, please add the missing \"const\"")
     virtual void GetSecondDerivativesVector(Vector& values, int Step = 0)
     {
-        if (values.size() != 0) {
-            values.resize(0, false);
-        }
+        const auto& r_const_this = *this;
+        r_const_this.GetSecondDerivativesVector(values, Step);
     }
```
Comment (rubenzorrilla) [inline kratos/includes/condition.h:375]: Same comment.

**Review by rubenzorrilla — DISMISSED** [review]
I went trough all the changes and I think they are all correct.

In any case this would be a compilation error.

File: kratos/includes/condition.h (hunk 1 above)
Comment (philbucher) [reply kratos/includes/condition.h:343]: nono this is just for backward compatibility

**Comment by philbucher** [comment]
@rubenzorrilla can you check again please?
I fixed the remaining errors

File: kratos/includes/condition.h (hunk 1 above)
Comment (rubenzorrilla) [reply kratos/includes/condition.h:343]: OK. Now I see.

**Review by rubenzorrilla — APPROVED** [review]

**Pull request merged**